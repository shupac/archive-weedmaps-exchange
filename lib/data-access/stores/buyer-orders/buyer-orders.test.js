import { getSnapshot } from 'mobx-state-tree';
import mockPurchaseOrder from 'lib/mocks/purchase-order';
import mockPurchaseOrders from 'lib/mocks/purchase-orders';
import BuyerOrders from './';

function setup(mockResp) {
  const mockFetchClient = {
    fetch: jest.fn().mockReturnValue(Promise.resolve({ data: mockResp })),
  };

  const buyerOrders = BuyerOrders.create(
    { orderData: mockPurchaseOrder },
    {
      client: mockFetchClient,
    },
  );
  return { buyerOrders, mockFetchClient };
}

describe('Buyer Orders Store', () => {
  it('can fetch purchase order and set data', async () => {
    const { buyerOrders } = setup(mockPurchaseOrder);
    await buyerOrders.fetchOrder();
    expect(getSnapshot(buyerOrders.orderData)).toMatchSnapshot();
  });
  it('can fetch all purchase orders and set data', async () => {
    const { buyerOrders } = setup(mockPurchaseOrders);
    await buyerOrders.fetchOrder();
    expect(getSnapshot(buyerOrders.ordersList)).toMatchSnapshot();
  });

  it('will handle errors', () => {
    const mockErrorClient = {
      fetch: () => {
        throw new Error('Test');
      },
    };
    const buyerOrders = BuyerOrders.create(
      { orderData: mockPurchaseOrder },
      {
        client: mockErrorClient,
      },
    );
    const log = jest.spyOn(global.console, 'error').mockReturnValue();
    buyerOrders.fetchOrder();
    buyerOrders.fetchPurchaseOrders();
    expect(log).toHaveBeenCalledWith(new Error('Test'));
    log.mockRestore();
  });
});
