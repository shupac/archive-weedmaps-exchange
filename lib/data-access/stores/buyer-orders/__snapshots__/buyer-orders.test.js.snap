// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buyer Orders Store can fetch all purchase orders and set data 1`] = `Array []`;

exports[`Buyer Orders Store can fetch purchase order and set data 1`] = `
Object {
  "buyerData": Object {
    "buyerAddress": "{\\"territory\\":\\"CA\\",\\"street_address\\":\\"55 Discovery\\",\\"region\\":{\\"name\\":\\"South OC Brands\\",\\"id\\":\\"64d05017-4339-4cda-9e57-0da061bf6b00\\"},\\"postal_code\\":\\"92618\\",\\"country\\":\\"us\\",\\"city\\":\\"Irvine\\"}",
    "buyerDeliveryInstructions": "TEST.",
    "buyerEmail": "asciametta@aol.com",
    "buyerLicenses": Array [
      Object {
        "id": "e5ae232c-9ae7-4e81-b6f6-67bd5562706f",
        "licenseType": "medical",
        "number": "THX1138",
      },
    ],
    "buyerLocationName": "Andrew's Cryb",
    "buyerPhone": "(303) 555-5555",
  },
  "expectedShipDateMax": "2018-11-15T17:48:29.169876Z",
  "expectedShipDateMin": "2018-11-13T17:48:29.169876Z",
  "givenDeliveryEta": Object {
    "etaMax": 6,
    "etaMaxUnit": "day",
    "etaMin": 4,
    "etaMinUnit": "day",
  },
  "id": "b97329d4-a7ae-4c7a-ab5e-4de8aec22f50",
  "orderDate": "2018-11-09T17:48:29.174071Z",
  "orderItems": Array [
    Object {
      "amount": 24,
      "avatarImage": Object {
        "id": "1d962557-b9bc-4e4a-9e16-a426820023be",
        "largeUrl": "https://images-acceptance.internal-weedmaps.com/products/000/035/422/avatar/large/wcc_gelatocan-12.jpg",
        "mediumUrl": "https://images-acceptance.internal-weedmaps.com/products/000/035/422/avatar/medium/wcc_gelatocan-12.jpg",
        "smallUrl": "https://images-acceptance.internal-weedmaps.com/products/000/035/422/avatar/square/wcc_gelatocan-12.jpg",
      },
      "categories": Object {
        "child": "Hybrid",
        "parent": "Flower",
      },
      "id": "2b113f61-1b7f-4d18-bd20-7c3ebc7d2007",
      "price": "100",
      "productId": "856c2fc1-854d-4eec-b62b-9d1670609110",
      "productName": "Gelato",
      "sku": "zzzzzzzz",
      "variantName": "1 ounces",
    },
    Object {
      "amount": 2,
      "avatarImage": Object {
        "id": "d636f9ff-411c-4d77-a32b-cf035ea77dd5",
        "largeUrl": "https://images-acceptance.internal-weedmaps.com/products/000/053/262/avatar/large/1507323767-171005_WestCoastCure_Lemon-Hard-Gelato-InsideShot-2072.jpg",
        "mediumUrl": "https://images-acceptance.internal-weedmaps.com/products/000/053/262/avatar/medium/1507323767-171005_WestCoastCure_Lemon-Hard-Gelato-InsideShot-2072.jpg",
        "smallUrl": "https://images-acceptance.internal-weedmaps.com/products/000/053/262/avatar/square/1507323767-171005_WestCoastCure_Lemon-Hard-Gelato-InsideShot-2072.jpg",
      },
      "categories": Object {
        "child": "Pre Rolls",
        "parent": "Flower",
      },
      "id": "7ada039c-fca7-462c-b87c-6d69a815d442",
      "price": "1000",
      "productId": "dbfa378b-85e6-44c0-a783-fe9650dc99f8",
      "productName": "Tahoe OG 3PC Cured Joints",
      "sku": "zzzzzzzz",
      "variantName": "50 Units",
    },
  ],
  "sellerData": Object {
    "sellerAddress": null,
    "sellerEmail": "test@test.com",
    "sellerId": "2f4ddb82-3b67-4780-8ae4-7f27da36d813",
    "sellerLicenses": Array [],
    "sellerName": "West Coast Cure",
    "sellerPhone": "1112223333",
  },
  "shippingFee": "0.0",
  "status": "not_started",
  "statusReason": "",
  "subtotal": "4400.0",
  "total": "4400",
}
`;
